### PR TITLE
feat(TDP-5987): Manage infinity values in dataSet cells

### PR DESCRIPTION
**Pull request by skermabon**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5987

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java [hunk 1]
```diff
@@ -382,4 +382,28 @@ public void test_ShouldHandleVeryLargeDoubleValue_TDQ_14216() {
         }
     }
 
+    @Test(timeout = 5000L)
+    public void shouldManageInfinity() {
```
Comment (aleboulanger) [inline dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java]: - Maybe add a reference to the JIRA issue

File: dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java [hunk 2]
```diff
@@ -382,4 +382,28 @@ public void test_ShouldHandleVeryLargeDoubleValue_TDQ_14216() {
...
+        // given
+        final StreamNumberHistogramStatistics histogram = new StreamNumberHistogramStatistics();
+        for (int i = 0; i <= 100; i++) {
```
Comment (aleboulanger) [inline dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java]: (_not a review_) you can use also `IntStream.range(0, 101).forEach(histogram::add)`

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatistics.java [hunk 3]
```diff
@@ -93,6 +93,10 @@
      * @param d the value to add to this histogram
      */
     public void add(double d) {
+        if (d == Double.POSITIVE_INFINITY || d == Double.NEGATIVE_INFINITY) {
```
[the remark this answers is not in the document]
Comment (skermabon) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatistics.java]: ok

File: dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java (hunk 1 above)
Comment (skermabon) [reply dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java]: ok

File: dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java (hunk 2 above)
Comment (skermabon) [reply dataprep-backend-common/src/test/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatisticsTest.java]: ok

**Pull request merged**